tests/integration/cmd_push_test.go

```diff
@@ -38,6 +38,17 @@ var _ = Describe("odo push command tests", func() {
 		os.Unsetenv("GLOBALODOCONFIG")
 	})
 
+	Context("Check pod timeout", func() {
+
+		It("Check that pod timeout works and we time out immediately..", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+			helper.CmdShouldPass("odo", "preference", "set", "PushTimeout", "1")
+			helper.CmdShouldFail("odo", "push", "--context", context+"/nodejs-ex")
```

Comment:
also check the error message like `waited 1s but couldn't find running pod matching selector` so that we are sure that the error is the intended one.